### ISSUE TITLE
Question about addr handling 

### ISSUE THREAD

**Issue by amitiuttarwar** [libbitcoin/libbitcoin-network#251]
Hi! I'm proposing some changes to addr handling in Bitcoin Core, and trying to ensure there will not be negative consequences for other clients on the network. For more information about the changes, please see my [mailing list post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-April/018784.html). The crux is that, if the changes are accepted, bitcoin core would stop sending addr messages to peers who don't "indicate interest" aka initiate an addr-related message (addr, addrv2, getaddr, or sendaddrv2).

I'd like to ensure that libbitcoin clients will initiate an addr related message to outbound peers. I've looked in the code and believe the software plays nice, and am looking for a +1 from someone more familiar with this project.

This is my understanding: in `session_outbound.cpp`, `attach_protocols` is called as part of the process of starting up a new outbound connection ([link](https://github.com/libbitcoin/libbitcoin-network/blob/197ce2b26a33f1d7c77d56ddf3f61cdfa7d6b53e/src/sessions/session_outbound.cpp#L133)). This function invokes `protocol_address_31402::start`, which sends a `getaddr` message ([link](https://github.com/libbitcoin/libbitcoin-network/blob/197ce2b26a33f1d7c77d56ddf3f61cdfa7d6b53e/src/protocols/protocol_address_31402.cpp#L78)). And it seems this is also the case for manual connections ([link](https://github.com/libbitcoin/libbitcoin-network/blob/197ce2b26a33f1d7c77d56ddf3f61cdfa7d6b53e/src/sessions/session_manual.cpp#L195)).

I also found that the starting sequence of `session_seed.cpp` invokes `protocol_seed_31402::start` ([link](https://github.com/libbitcoin/libbitcoin-network/blob/197ce2b26a33f1d7c77d56ddf3f61cdfa7d6b53e/src/sessions/session_seed.cpp#L216)), which looks like it sends a `getaddr` and self advertises its own address ([link](https://github.com/libbitcoin/libbitcoin-network/blob/197ce2b26a33f1d7c77d56ddf3f61cdfa7d6b53e/src/protocols/protocol_seed_31402.cpp#L66-L85)). 

Although my understanding is incomplete (does the node self announce to seeds, but not to normal outbound / manual connections?), if my claims above are accurate, there should be no compatibility issues with my proposed changes. 

tldr; I'm proposing a change to Bitcoin Core and want to make sure it does not break libbitcoin clients, can someone confirm my understanding of the code?

Thanks in advance!
Amiti

**Comment by evoskuil** [comment]
The node advertises itself on all sessions where the address protocol is attached (default for all three types).

The seeding session is distinct because of a limited set of attached protocols, and a specific timeout on connections.  The only other distinction is that seed nodes (any node) are listed in config.

Thanks for the heads up.

**Comment by amitiuttarwar** [comment]
awesome, thank you for confirming!
